### PR TITLE
[BUG] Fix search topbar active and hover state, pagination buttons, panel header

### PR DESCRIPTION
**Pull request by chth0n1x**
Tickets (Notion): 
    -https://www.notion.so/cos/Pagination-buttons-added-styling-0ef0c6bfca96423e9b323fd7b9031172?pvs=4
    -https://www.notion.so/cos/Selenium-Request-to-add-data-test-attributes-to-Search-Page-Elements-969958ae9c1e431a98125584bd8c6028?pvs=4
    
GitHub branch: feature/search-improvements-b-and-i-part-3

## Purpose

The purpose of these changes was to fix the active and hover states of the search top bar, the excessive top margin in the left panel header, increase pagination button padding, and to provide QA with data test selectors for the result card container and its individual title link.

## Summary of Changes

- Top margin of the left panel header was reduced to balance it with the corresponding facet filter border
- Width of list elements of the topbar was evenly divided across topbar width
- Padding was added to the pagination buttons to improve visibility
- Data test selectors were added to the result card and the result card title link

## Screenshot(s)

image

Figure 1: Topbar with corrected active and hover states

image

Figure 2: Pagination buttons with proper spacing, shape, and size

## Side Effects

This will affect the overall UI of the search topbar, facet filter left panel, and the pagination button section. Testing should include changes to it and the surrounding layout.

## QA Notes

1. Does the UI look like the mock ups? Should anything be added/removed: https://preview.uxpin.com/e2821be91003954dec0c29a026bc49989b868e2d#/pages/162679105/simulate/no-panels?mode=i

### CODE REVIEW

File: lib/osf-components/addon/components/search-page/template.hbs [hunk 1]
```diff
@@ -241,25 +241,38 @@ as |layout|>
...
+                </div>
+            {{/if}}
             {{#if this.nextPageCursor}}
                 <Button
                     {{on 'click' (fn this.switchPage this.nextPageCursor)}}
                 >
-                    {{t 'search.next'}}
+                    <p>
+                        {{t 'search.next-pagination-button'}}
+                    </p>
```
Comment (adlius) [inline lib/osf-components/addon/components/search-page/template.hbs:275]: During one of the scrum discussion, we have decided to have three pagination buttons (First, Prev, Last). So it is okay to not follow the mockup here.

File: translations/en-us.yml [hunk 2]
```diff
@@ -246,6 +246,10 @@ search:
     first: First
     prev: Prev
     next: Next
+    previous-pagination-button: '<<'
+    first-pagination-button: '1'
+    more-pagination-button: '...'
+    next-pagination-button: '>>'
```
Comment (adlius) [inline translations/en-us.yml:252]: We should revert these changes.

File: lib/osf-components/addon/components/search-page/styles.scss [hunk 3]
```diff
@@ -366,8 +369,33 @@
...
+    > div {
+        background-color: $color-bg-white;
+        color: $color-bg-blue-dark;
+        display: flex;
+        justify-content: center;
+        align-items: center;
+        height: 44px;
+        width: 44px;
+        text-align: center;
+    }
```
Comment (adlius) [inline lib/osf-components/addon/components/search-page/styles.scss:398]: This is not needed if we don't have the ellipsis.

**Review by adlius — CHANGES_REQUESTED** [review]

File: lib/osf-components/addon/components/search-page/template.hbs [hunk 4]
```diff
@@ -241,25 +241,38 @@ as |layout|>
...
         {{/if}}
         <div local-class='pagination-buttons'>
-            {{#if this.firstPageCursor}}
+            {{#if this.prevPageCursor}}
                 <Button
-                    {{on 'click' (fn this.switchPage this.firstPageCursor)}}
+                    {{on 'click' (fn this.switchPage this.prevPageCursor)}}
                 >
-                    {{t 'search.first'}}
+                    <p>
```
[the remark this answers is not in the document]
Comment (chth0n1x) [reply lib/osf-components/addon/components/search-page/template.hbs:248]: Ok, this is now removed.

File: lib/osf-components/addon/components/search-page/template.hbs [hunk 5]
```diff
@@ -241,25 +241,38 @@ as |layout|>
...
                 </Button>
             {{/if}}
-            {{#if this.prevPageCursor}}
+            {{#if this.firstPageCursor}}
                 <Button
-                    {{on 'click' (fn this.switchPage this.prevPageCursor)}}
+                    {{on 'click' (fn this.switchPage this.firstPageCursor)}}
                 >
-                    {{t 'search.prev'}}
+                    <p>
```
[the remark this answers is not in the document]
Comment (chth0n1x) [reply lib/osf-components/addon/components/search-page/template.hbs:257]: Removed here as well.

File: lib/osf-components/addon/components/search-page/template.hbs [hunk 6]
```diff
@@ -241,25 +241,38 @@ as |layout|>
...
+                        {{t 'search.first-pagination-button'}}
+                    </p>
                 </Button>
             {{/if}}
+            {{#if this.nextPageCursor}}
+                <div>
+                    <p>
```
[the remark this answers is not in the document]
Comment (chth0n1x) [reply lib/osf-components/addon/components/search-page/template.hbs:264]: Removed here as well.

File: lib/osf-components/addon/components/search-page/template.hbs (hunk 1 above)
Comment (chth0n1x) [reply lib/osf-components/addon/components/search-page/template.hbs:275]: Ok, that sounds good.

File: translations/en-us.yml (hunk 2 above)
Comment (chth0n1x) [reply translations/en-us.yml:252]: These are now reverted.

File: lib/osf-components/addon/components/search-page/styles.scss (hunk 3 above)
Comment (chth0n1x) [reply lib/osf-components/addon/components/search-page/styles.scss]: Ok, unnecessary styling is now removed. Background color and the left margin for the pagination buttons have been kept to position and add color to the button text.

**Review by adlius — APPROVED** [review]
